lib/Conversion/Utils/Utils.cpp

```diff
@@ -235,20 +235,24 @@ Value convertScalarToDtype(OpBuilder &b, Location loc, Value scalar,
   if (scalarType == dtype)
     return scalar;
 
-  // TODO: For the byte(ui8) or char(i8) case, we need the unconverted dtype to
-  // be able to know if we need signed or unsigned conversion.
+  // TODO: For the conversion to byte(ui8) or char(i8) case, we need the
+  // unconverted dtype to be able to know if we need signed or unsigned
+  // conversion.
   auto isByteOrChar = [](Type type) {
     if (auto integerTy = type.dyn_cast<mlir::IntegerType>()) {
       return integerTy.getWidth() == 8;
     }
     return false;
   };
 
-  if (isByteOrChar(scalarType) || isByteOrChar(dtype)) {
+  // We only support conversion from Byte or Char scalarType not to Byte or Char
+  // dtype.
+  if (isByteOrChar(dtype)) {
```

Comment:
This isn't the right solution. The code below makes the assumption that `ui8` will not make it past this point:

https://github.com/llvm/torch-mlir/blob/a329356fbfce6f17ecf6899a32a7d623812bb8b4/lib/Conversion/Utils/Utils.cpp#L302-L303

Also see TODO: 

https://github.com/llvm/torch-mlir/blob/a329356fbfce6f17ecf6899a32a7d623812bb8b4/lib/Conversion/Utils/Utils.cpp#L238